### DISCUSSION TITLE
Routing like next.js

### DISCUSSION THREAD

**Discussion by kevinsimper** [trpc/trpc#3366] [Ideas]
Next.js has a big appeal with the file-system based router.
https://nextjs.org/docs/routing/introduction

It makes it easy to find code and delete code. All code is colocated next to the route itself.

It also has its downsides like it is hard to spread the code when the route grows without loosing the benefit of colocated together in the first place.

GraphQL compared to trpc makes it even harder to know where a certain query or mutation is used.

However trpc has potentially the problem of a query being used across multiple pages, which means updating that query affects multiple pages.
Also weirdly specific queries/mutations like on-off queries for dashboards or specific backoffice functions can populate the trpc graph. These are then forgotten to be removed when the dashboard or backoffice page is inevitable deleted.

So would routing like next.js make sense? Like where the specific backoffice function lives next to the next.js page where it is used?

**Comment by KATT** [comment]
How would it practically look like?

With the router structure we have now, it'd be pretty simple to create a watcher that stitches together a folder structure of procedures or routers automatically

**Reply by kevinsimper** [reply]
That was also something I considered, but that looks like graphql-codegen that one wanted to get away from.

Otherwise it could be something that ts-prune could be interesting
https://github.com/nadeesha/ts-prune
